libsolidity/formal/SMTChecker.cpp

```diff
@@ -853,9 +896,41 @@ void SMTChecker::arithmeticOperation(BinaryOperation const& _op)
 			m_interface->addAssertion(right != 0);
 		}
 
-		checkUnderOverflow(value, intType, _op.location());
+		addOverflowTarget(
+			OverflowTarget::Type::All,
+			_op.annotation().commonType,
+			value,
+			_op.location()
+		);
 
 		defineExpr(_op, value);
+
+		smt::Expression powerOf2 = (0 - minValue(intType)) + maxValue(intType) + 1;
+		if (op == Token::Add || op == Token::Mul)
+		{
+			defineExpr(_op, smt::Expression::ite(
+				value > maxValue(intType),
+				value % powerOf2,
+				value
+			));
+			if (intType.isSigned())
+			{
+				defineExpr(_op, smt::Expression::ite(
+					expr(_op) > maxValue(intType),
+					expr(_op) - powerOf2,
+					expr(_op)
+				));
+			}
+		}
+		else if (op == Token::Sub)
+		{
+			defineExpr(_op, smt::Expression::ite(
+				value < minValue(intType),
+				value % powerOf2,
```

Comment:
So for ``value = -129``... this is ``-129 % 256``... is this always ``127``?
Can you instead do ``value + powerOf2`` to avoid negative modulos?
I need to have another look at this in general...

Reply:
This can probably be ignored - nobody will use negative representatives for remainders.